app/services/music_import_service/recording_collector.rb

```diff
@@ -0,0 +1,334 @@
+  def initialize(sql_server_adapter:, postgres_adapter:, logger:, cache: MarshalCache.new("tmp"))
+    @sql_server_adapter = sql_server_adapter
+    @postgres_adapter = postgres_adapter
+    @logger = logger
+    @cache = cache || NullCache
+  end
+
+  class MarshalCache
+    attr_reader :location
+    def initialize(location)
+      @location = Pathname.new(location)
+    end
+
+    def fetch(file, default = nil)
+      if File.exist?(location.join(file))
+        Marshal.load(File.open(location.join(file)))
+      elsif block_given?
+        results = yield
+        store(file, results)
+      else
+        default
+      end
+    end
+
+    def store(file, value)
+      FileUtils.mkdir_p(location)
+      File.open(location.join(file), "wb") do |f|
+        f.puts Marshal.dump(value)
+      end
+      value
+    end
+  end
+
+  class NullCache
+    def self.fetch(_file, default = nil)
+      if block_given?
+        yield
+      else
+        default
+      end
+    end
+
+    def self.store(_file, value)
+      value
+    end
+  end
+
+  def recordings
+    @recordings ||=
+      begin
+        cache.fetch("recordings_cache.dump") do
+          logger.info "loading recordings"
+          results = sql_server_adapter.execute(query: recordings_query).group_by { |result| result["idRecording"] }
+          results = results.map do |result|
+            MRRecording.new(result.first,
+                            result.second.first["CallNo"], # call number comes from recording so is same for each entry
+                            result.second.map { |t| t["CourseNo"] }.uniq.compact, # comes from joins
+                            result.second.map { |t| t["RecTitle"] }.uniq.compact)
+          end
+          logger.info "reconciling callnumbers to bibids"
+          results.each do |rec|
+            logger.info "reconciling call number for #{rec}"
+            rec.bibs = bib_numbers_for(call_number: rec.call)
+            logger.info "  got #{rec.bibs}"
+          end
+          # Find bibs via title
+          @found_bibs = 0
+          results.select { |x| x.bibs.empty? }.each do |recording|
+            populate_bib_from_title(recording)
+          end
+          # Find "correct" bib via Leveshtein distance.
+          results.select { |x| x.bibs.length > 1 }.each do |recording|
+            bib_records = bib_records_from_recording(recording).select(&:present?)
+            recording.recommended_bib = bib_records.find { |x| x[:title_distance] <= 6 }.try(:[], :id)
+          end
+          store_cached_bibs
+          logger.info "Found #{@found_bibs} bibs from searching"
+          results
+        end
+      end
+  end
+
+  def populate_bib_from_title(recording)
+    logger.info "populating bib from title for #{recording.titles.first}"
+    ol_response = JSON.parse(
+      open(
+        "https://catalog-staging.princeton.edu/catalog.json?f[access_facet][]=In+the+Library&f[format][]=Audio&f[location][]=Mendel+Music+Library" \
```

Comment:
The other URLs are the production URLs — change this one to match?

Reply:
or should this be configurable? Or keyed off of the selected db? @tpendragon

Reply:
Configurable seems easiest

Reply:
@tpendragon like add a param to the initializer or use an actual config file?

Reply:
A param in the initializer seems right.

Reply:
okay great i agree. i'll take care of it

Reply:
defaulted it to production.